tests/src/receive_spec.cpp

```diff
@@ -242,6 +279,7 @@ int main()
     test_receive_stream();
     test_receive_max_sized_message();
     test_receive_oversized_message();
+    test_resize_buffer();
```

Comment:
Since the merge request contains functionality to dynamically change the buffer size it would be suitable to have tests that verify that this works well. E g:

- What happens if setBufferSize() fails to allocate the memory?
- If the buffer size is shrunk will it fail (appropriately) to receive a message that was within the prior limits?
- And the reverse, if it fails to receive a message due to size will increasing the buffer size accordingly help?

Reply:
yes, all make sense. Let's see if this is likely to be merged first.